lib/mosql/streamer.rb

```diff
@@ -73,13 +73,30 @@ def with_retries(tries=10)
           raise if e.kind_of?(Mongo::OperationFailure) && [11000, 11001].include?(e.error_code)
           # Cursor timeout
           raise if e.kind_of?(Mongo::OperationFailure) && e.message =~ /^Query response returned CURSOR_NOT_FOUND/
-          delay = 0.5 * (1.5 ** try)
+          delay = 0.5 * (2 ** try)
           log.warn("Mongo exception: #{e}, sleeping #{delay}s...")
           sleep(delay)
         end
       end
     end
 
+    def tail_with_retries(tries=3)
+      tries.times do |try|
+        begin
+          yield
+        rescue Sequel::PoolTimeout, Sequel::DatabaseConnectionError, Sequel::DatabaseDisconnectError => e
+          # Be less aggressive with the rerties. Every ten minutes three times.
+          delay = (10 * 60) * (try + 1)
```

Comment:
A ten minute initial delay seems pretty rough; it'd actually move us *way* backward since a network blip currently only causes a few seconds of delay (long enough for eye to restart the crashed process).

Exponential backoff is probably a better solution.